pkg/products/threescale/reconciler.go

```diff
@@ -1379,30 +1494,149 @@ func (r *Reconciler) reconcile3scaleMultiTenancy(ctx context.Context, serverClie
 	return integreatlyv1alpha1.PhaseCompleted, nil
 }
 
-func getMTAccountsToBeCreated(usersIdentity []usersv1.Identity, accounts []Account) []Account {
-	accountsToBeCreated := []Account{}
+func (r *Reconciler) reconcileDashboardLink(ctx context.Context, serverClient k8sclient.Client, username string, tenantLink string) error {
+	cl := &consolev1.ConsoleLink{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: username + "-3scale",
+		},
+	}
+
+	tenantNamespaces := []string{fmt.Sprintf("%s-stage", username), fmt.Sprintf("%s-dev", username)}
+	_, err := controllerutil.CreateOrUpdate(ctx, serverClient, cl, func() error {
+		cl.Spec = consolev1.ConsoleLinkSpec{
+			Location: consolev1.NamespaceDashboard,
+			Link: consolev1.Link{
+				Href: tenantLink,
+				Text: "API Management",
+			},
+			NamespaceDashboard: &consolev1.NamespaceDashboardSpec{
+				Namespaces: tenantNamespaces,
+			},
+		}
+
+		return nil
+	})
+	if err != nil {
+		return fmt.Errorf("error reconciling console link: %v", err)
+	}
+
+	return nil
+}
+
+func (r *Reconciler) AddAuthProviderToMTAccount(ctx context.Context, serverClient k8sclient.Client, account SignUpAccount) error {
+
+	tenantID := string(account.AccountDetail.OrgName)
+	clientID := fmt.Sprintf("%s-%s", multitenantID, tenantID)
+	integration := fmt.Sprintf("%s-%s", rhssoIntegrationName, clientID)
+
+	isAdded, err := r.tsClient.IsAuthProviderAdded(account.AccountAccessToken.Value,
+		integration, account.AccountDetail)
+	if err != nil {
+		return err
+	}
+	if isAdded {
+		return nil
+	}
+
+	oauthClientSecrets := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "tenant-oauth-client-secrets",
+			Namespace: r.installation.GetNamespace(),
+		},
+	}
+
+	err = serverClient.Get(ctx, k8sclient.ObjectKey{
+		Name:      oauthClientSecrets.Name,
+		Namespace: oauthClientSecrets.Namespace},
+		oauthClientSecrets,
+	)
+	if err != nil {
+		r.log.Errorf("could not find secret", l.Fields{"secret": oauthClientSecrets.Name, "operatorNamespace": oauthClientSecrets.Namespace}, err)
+		return fmt.Errorf("could not find %s Secret: %w", oauthClientSecrets.Name, err)
+	}
+
+	secret, ok := oauthClientSecrets.Data[tenantID]
```

Comment:
Could data be nil here?

Reply:
it could but as the creation of the secret is handled in the bootstrap stage https://github.com/integr8ly/integreatly-operator/blob/master/controllers/rhmi/bootstrapReconciler.go#L312 and it makes sure data is created I think we don't need to verify it here